### PR TITLE
Bs 51

### LINKED ISSUE #51: Написать сервис обработки in app purchase & restore purchase (RevenueCat)

### PR DESCRIPTION
**Pull request by antonhoang**

### CODE REVIEW

File: bonsai/Services/StoreService/PurchaseService.swift [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+   private func checkIfUserSubscriptionStatus() {
+      Purchases.shared.getCustomerInfo { customerInfo, error in
+         self.isSubscriptionActive = customerInfo?.entitlements.all[Pro.typeName]?.isActive == true
+      }
+   }
+   
+   func buy(package: Package?) {
+      if let package = package {
+         Purchases.shared.purchase(package: package) { storeTransaction, customerInfo, error, bool in
+            if let error = error {
```
Comment (VladimirBrejcha) [inline bonsai/Services/StoreService/PurchaseService.swift:47]: Lets add at least asserts on missing error handlers

File: bonsai/Services/StoreService/PurchaseService.swift [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            }
+            if let storeTransaction = storeTransaction {
+            }
+         }
+      }
+   }
+   
+   func restorePurchase() {
+      Purchases.shared.restorePurchases { customerInfo, error in
+         if let error = error {
```
Comment (VladimirBrejcha) [inline bonsai/Services/StoreService/PurchaseService.swift:60]: same

File: bonsai/Services/StoreService/PurchaseService.swift [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+         }
+         if let allEntitlements = customerInfo?.entitlements.all[Pro.typeName] {
+            self.isSubscriptionActive = allEntitlements.isActive
+         } else {
+            // send an alert
```
Comment (VladimirBrejcha) [inline bonsai/Services/StoreService/PurchaseService.swift]: send an alert

File: bonsai/bonsaiApp.swift [hunk 4]
```diff
@@ -6,17 +6,26 @@
...
          ContentView()
             .environment(\.managedObjectContext, dataController.container.viewContext)
             .environment(\.persistentContainer, dataController.container)
+            .environmentObject(purchaseService)
       }
    }
+   
+   init() {
+      Purchases.logLevel = .debug
+      Purchases.configure(withAPIKey: "appl_ccwfbkcrUyDDSxBwwExoLwsgGya")
```
Comment (VladimirBrejcha) [inline bonsai/bonsaiApp.swift]: api key to plist

File: bonsai/Services/StoreService/PurchaseService.swift (hunk 3 above)
Comment (antonhoang) [reply bonsai/Services/StoreService/PurchaseService.swift]: +

File: bonsai/Services/StoreService/PurchaseService.swift (hunk 2 above)
Comment (antonhoang) [reply bonsai/Services/StoreService/PurchaseService.swift:60]: +

File: bonsai/Services/StoreService/PurchaseService.swift (hunk 1 above)
Comment (antonhoang) [reply bonsai/Services/StoreService/PurchaseService.swift:47]: +

**Pull request merged**